pkg/kv/kvserver/replica_circuit_breaker_cancelstorage.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package kvserver
+
+import (
+	"context"
+	"sync"
+	"unsafe"
+
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+)
+
+// CancelStorage implements tracking of context cancellation functions
+// for use by Replica circuit breakers.
+type CancelStorage interface {
```

Comment:
This seems mostly self-explanatory, but very brief method comments could be useful.